plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftPublishController.java

```diff
@@ -186,4 +199,33 @@ protected boolean isEapProfile() {
 		String profile = ServerProfileModel.getProfile(getServer());
 		return OpenshiftEapProfileDetector.PROFILE.equals(profile);
 	}
+
+	public void deleteOldDeployments(final IServer server) throws CoreException {
+		File deploymentsRootFolder = getDeploymentsRootFolder();
+		Arrays.stream(server.getModules())
+			.forEach(module -> deleteOldDeployment(module, deploymentsRootFolder));
```

Comment:
Seems the output name cache should be reset here, not below.

Reply:
yes that's where individual resets should happen.